crates/boundless-market/src/dynamic_gas_filler.rs

```diff
@@ -12,57 +12,228 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::sync::Arc;
+
 use alloy::{
-    network::{Network, TransactionBuilder},
+    consensus::BlockHeader,
+    eips::{eip1559::Eip1559Estimation, BlockNumberOrTag},
+    network::{BlockResponse, Network, TransactionBuilder},
     primitives::Address,
     providers::{
         fillers::{FillerControlFlow, GasFillable, GasFiller, TxFiller},
-        Provider, SendableTx,
+        utils::{self, Eip1559Estimator, Eip1559EstimatorFn},
+        Provider, RootProvider, SendableTx,
     },
-    transports::TransportResult,
+    transports::{RpcError, TransportResult},
 };
+use serde::{Deserialize, Serialize};
+use tokio::sync::RwLock;
+
+const DEFAULT_FEE_HISTORY_PERCENTILES: [f64; 1] = [utils::EIP1559_FEE_ESTIMATION_REWARD_PERCENTILE];
+const LOW_PRIORITY_PERCENTILES: [f64; 1] = [10.0];
+const MEDIUM_PRIORITY_PERCENTILES: [f64; 1] = [20.0];
+const HIGH_PRIORITY_PERCENTILES: [f64; 1] = [30.0];
+const DEFAULT_BASE_FEE_MULTIPLIER_PERCENTAGE: u64 =
+    (utils::EIP1559_BASE_FEE_MULTIPLIER as u64) * 100;
+
+/// Priority mode for transaction gas pricing.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize, Default)]
+#[serde(rename_all = "snake_case")]
+#[non_exhaustive]
+pub enum PriorityMode {
+    /// Use the eth_feeHistory as is and only add `+3%` per pending tx.
+    Low,
+    /// Add a `+20%` increase to the base/priority fee and `+5%` per pending transaction.
```

Comment:
Comment looks incorrect here and below, think you mean use 20th percentile etc

Reply:
yes, was from previous version. https://github.com/boundless-xyz/boundless/pull/1339/commits/d2d888e8457fa453e9ab94b1abbb72b59b00f6ae fixed, thanks